src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -158,6 +161,105 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
     return Status(StatusCode::INTERNAL_ERROR, "Unexpected error during Tensor creation");
 }
 
+static mediapipe::ImageFormat::Format KFSDatatypeToImageFormat(const std::string& datatype, const size_t& numberOfChannels) {
+    if(datatype == "UINT8" || datatype == "INT8")
+    {
+        if(numberOfChannels == 1){
+            return mediapipe::ImageFormat::GRAY8;
+        }
+        if(numberOfChannels == 3){
+            return mediapipe::ImageFormat::SRGB;
+        }
+        if(numberOfChannels == 4){
+            return mediapipe::ImageFormat::SRGBA;
+        }
+    }
+    if(datatype == "UINT16" || datatype == "INT16")
+    {
+        if(numberOfChannels == 1){
+            return mediapipe::ImageFormat::GRAY16;
+        }
+        if(numberOfChannels == 3){
+            return mediapipe::ImageFormat::SRGB48;
+        }
+        if(numberOfChannels == 4){
+            return mediapipe::ImageFormat::SRGBA64;
+        }
+    }
+    if(datatype == "FP16")
+    {
+        if(numberOfChannels == 1){
+            return mediapipe::ImageFormat::GRAY16;
+        }
+        if(numberOfChannels == 3){
+            return mediapipe::ImageFormat::SRGB48;
+        }
+        if(numberOfChannels == 4){
+            return mediapipe::ImageFormat::SRGBA64;
+        }
+    }
+    if(datatype == "FP32")
```

Comment:
Minor: this could be at the top of the function, this is the most used precision I guess